base/src/main/java/com/austindroids/austinfeedsme/eventsfilter/EventFilterPresenter.java

```diff
@@ -1,52 +1,87 @@
 package com.austindroids.austinfeedsme.eventsfilter;
 
 import com.austindroids.austinfeedsme.data.Event;
-import com.austindroids.austinfeedsme.data.EventsDataSource;
+import com.austindroids.austinfeedsme.data.EventsRepository;
+import com.austindroids.austinfeedsme.data.FilterableEventsRepository;
+import com.austindroids.austinfeedsme.di.modules.DataModule.Eventbrite;
+import com.austindroids.austinfeedsme.di.modules.DataModule.Meetup;
 
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.functions.Consumer;
 
 /**
  * Created by darrankelinske on 8/4/16.
  */
 public class EventFilterPresenter implements EventFilterContract.Presenter {
 
-    private static final String TAG ="EventFilterPresenter";
-
-    private EventsDataSource eventbriteRepository;
-    private EventsDataSource meetupRepository;
+    private FilterableEventsRepository filterableEventsRepository;
+    private EventsRepository eventbriteRepository;
+    private EventsRepository meetupRepository;
     private EventFilterContract.View view;
 
-    public EventFilterPresenter(EventsDataSource eventsDataSource,
-                                EventsDataSource meetupDataSource, EventFilterContract.View view) {
-        this.eventbriteRepository = eventsDataSource;
-        this.meetupRepository = meetupDataSource;
+    @Inject
+    public EventFilterPresenter(FilterableEventsRepository filterableEventsRepository, @Eventbrite EventsRepository eventbriteRepository,
+                                @Meetup EventsRepository meetupRepository, EventFilterContract.View view) {
+        this.filterableEventsRepository = filterableEventsRepository;
+        this.eventbriteRepository = eventbriteRepository;
+        this.meetupRepository = meetupRepository;
         this.view = view;
     }
 
     @Override
     public void loadEvents() {
-        eventbriteRepository.getEvents(new EventsDataSource.LoadEventsCallback() {
-            @Override
-            public void onEventsLoaded(List<Event> events) {
-                view.showEvents(events);
-            }
 
-            @Override
-            public void onError(String error) {
+        Observable<List<Event>> mergedIncomingEvents = eventbriteRepository.getEventsRX();
 
-            }
-        }, false);
+        Observable.combineLatest(filterableEventsRepository.getEventsRX(true, false), mergedIncomingEvents,
```

Comment:
Hmm, who's going to dispose this later, and when? Should this actually be `Single.zip()`?